### PR TITLE
Added LCS implementation in Ruby (#2252)

### LINKED ISSUE #2252: Longest Common Subsequence
- [ ] C - @NVS16
- [x] C++
- [ ] C# - @Yashank18
- [ ] CoffeeScript
- [ ] Dart - @JainStuti25
- [x] Go
- [x] Java
- [x] JS - @abhu-A-J
- [ ] Kotlin - @rbjoshi1309
- [ ] Php - @thakran14
- [x] Python
- [ ] Readme - @NikitaMasand
- [x] Ruby - @Tahanima
- [ ] Rust
- [ ] TypeScript - @adi44

### PR DESCRIPTION
**Pull request by Tahanima**
Fixes #2252 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added implementation of Longest Common Subsequence in Ruby

#### Languages Used:
- Ruby

#### Files Added:
- Added Longest_Common_Subsequence.rb under Longeest_Common_Subsequence folder

Thanks!

### CODE REVIEW

**Comment by Tahanima** [comment]
Hello @jainaman224 

I have implemented LCS algorithm in Ruby. Please review the changes and let me know if I need to add/change something.

Thank You

File: Longest_Common_Subsequence/Longest_Common_Subsequence.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                lcs[i][j] = [lcs[i - 1][j], lcs[i][j - 1]].max
+            end
+        end 
+    end
+    
+    lcs[size1][size2]
+end 
+
+# Driver Code
+arr1 = [10, 15, 20, 25, 30, 35, 40]
```
Comment (singh-shreya6) [inline Longest_Common_Subsequence/Longest_Common_Subsequence.rb:31]: It is possible to take dynamic inputs in Ruby. Please update the code to take dynamic inputs. This requirement is only relaxed for languages like js or php where we have to create forms for dynamic inputs.

File: Longest_Common_Subsequence/Longest_Common_Subsequence.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+arr2 = [10, 12, 23, 25, 28, 30, 32, 40]
+size1 = arr1.length()
+size2 = arr2.length()
+len = longest_common_subsequence(arr1, arr2, size1, size2)
+
+# The longest common subsequence comprises of elements 10, 25, 30, 40 in the above arrays
+# Output:
+# Length of Longest Common Subsequence is: 4
+puts "Length of Longest Common Subsequence is: #{len}"
```
Comment (singh-shreya6) [inline Longest_Common_Subsequence/Longest_Common_Subsequence.rb:40]: Please add a blank line at end of file

File: Longest_Common_Subsequence/Longest_Common_Subsequence.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# Problem Statement- Given two sequences, find
```
Comment (singh-shreya6) [inline Longest_Common_Subsequence/Longest_Common_Subsequence.rb:1]: Please convert these to a single multi line comment.

File: Longest_Common_Subsequence/Longest_Common_Subsequence.rb (hunk 1 above)
Comment (Tahanima) [reply Longest_Common_Subsequence/Longest_Common_Subsequence.rb]: Have changed to dynamic input

File: Longest_Common_Subsequence/Longest_Common_Subsequence.rb (hunk 2 above)
Comment (Tahanima) [reply Longest_Common_Subsequence/Longest_Common_Subsequence.rb]: Added blank line

File: Longest_Common_Subsequence/Longest_Common_Subsequence.rb (hunk 3 above)
Comment (Tahanima) [reply Longest_Common_Subsequence/Longest_Common_Subsequence.rb]: Converted them to a single multi-line comment

File: Longest_Common_Subsequence/Longest_Common_Subsequence.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        end 
+    end
+    
+    lcs[size1][size2]
+end 
+
+# Driver Code
+puts "Enter the size of first array:"
+size1 = gets.to_i
+puts "Enter the first array elements:"
```
Comment (singh-shreya6) [inline Longest_Common_Subsequence/Longest_Common_Subsequence.rb:35]: Please add one blank line before this for readability.

File: Longest_Common_Subsequence/Longest_Common_Subsequence.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+arr1 = gets.split(" ").map(&:to_i)
+puts "Enter the size of second array:"
```
Comment (singh-shreya6) [inline Longest_Common_Subsequence/Longest_Common_Subsequence.rb:37]: Please add one blank line before this for readability.

File: Longest_Common_Subsequence/Longest_Common_Subsequence.rb [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+size2 = gets.to_i
+puts "Enter the second array elements:"
```
Comment (singh-shreya6) [inline Longest_Common_Subsequence/Longest_Common_Subsequence.rb:39]: Please add one blank line before this for readability.

File: Longest_Common_Subsequence/Longest_Common_Subsequence.rb (hunk 4 above)
Comment (Tahanima) [reply Longest_Common_Subsequence/Longest_Common_Subsequence.rb:36]: Done

File: Longest_Common_Subsequence/Longest_Common_Subsequence.rb (hunk 5 above)
Comment (Tahanima) [reply Longest_Common_Subsequence/Longest_Common_Subsequence.rb:39]: Done

File: Longest_Common_Subsequence/Longest_Common_Subsequence.rb (hunk 6 above)
Comment (Tahanima) [reply Longest_Common_Subsequence/Longest_Common_Subsequence.rb:42]: Done

**Pull request merged**